oss/node-express-spawn/app.js

```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+const {spawn} = require('child_process');
+const express = require('express');
+const app = express();
+
+const {PORT = 3000} = process.env;
+
+app.get('/dir/*', function (req, res) {
+  spawnPgm('ls', ['-lh', req.path.split('/dir')[1]], (code, output) => {
+    res.set('Content-Type', 'text/plain');
+    res.send(output);
+  });
+});
+
+app.get('/cmd/*', function (req, res) {
+  const pgm = req.path.split('/cmd/')[1];
+  const args = req.query.args ? req.query.args.split(',') : [];
+  //
+  spawnPgm(pgm, args, (code, output) => {
+    res.set('Content-Type', 'text/plain');
+    res.send(output);
+  });
+});
+
+function spawnPgm(pgm, args, done) {
+  const run = spawn(pgm, args);
+
+  let output = '';
+  run.stdout.on('data', (data) => {
+    output += data + '\n';
```

Comment:
hmm actually I'm not sure it can assume each chunk is a line, but it'd probably be fine to remove \n's since the command output will have its own newlines